toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -72,43 +92,508 @@ def send_dm_to_user(
+            "Error sending message",
+            developer_message=f"Slack API Error: {error_message}",
+        )
+    else:
+        return {"response": response.data}
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["channels:read", "groups:read", "im:read", "mpim:read"],
     )
+)
+async def get_members_from_conversation_by_id(
+    context: ToolContext,
+    conversation_id: Annotated[str, "The ID of the conversation to get members for"],
+    limit: Annotated[
+        Optional[int], "The maximum number of members to return."
+    ] = MAX_PAGINATION_SIZE_LIMIT,
+    next_cursor: Annotated[Optional[str], "The cursor to use for pagination."] = None,
+) -> Annotated[dict, "Information about each member in the conversation"]:
+    """Get the members of a conversation in Slack by the conversation's ID."""
+    slackClient = AsyncWebClient(token=context.authorization.token)
 
-    # Step 1: Retrieve the list of channels
-    channels_response = slackClient.conversations_list()
-    channel_id = None
-    for channel in channels_response["channels"]:
-        if channel["name"].lower() == channel_name.lower():
-            channel_id = channel["id"]
-            break
+    try:
+        member_ids, next_cursor = await async_paginate(
+            slackClient.conversations_members,
+            "members",
+            limit=limit,
+            next_cursor=next_cursor,
+            channel=conversation_id,
+        )
+    except SlackApiError as e:
+        if e.response["error"] == "channel_not_found":
+            conversations = await list_conversations_metadata(context)
+            available_conversations = ", ".join(
+                f"{conversation['id']} ({conversation['name']})"
+                for conversation in conversations["conversations"]
+            )
 
-    if not channel_id:
-        raise RetryableToolError(
-            "Channel not found",
-            developer_message=f"Channel with name '{channel_name}' not found.",
-            additional_prompt_content=format_channels(channels_response),
-            retry_after_ms=500,  # Play nice with Slack API rate limits
+            raise RetryableToolError(
+                "Conversation not found",
+                developer_message=f"Conversation with ID '{conversation_id}' not found.",
+                additional_prompt_content=f"Available conversations: {available_conversations}",
+                retry_after_ms=500,
+            )
+
+    # Get the members' info
+    # TODO: This will probably hit rate limits. We should probably call list_users() and
+    # then filter the results instead.
+    members = await asyncio.gather(*[
+        get_user_info_by_id(context, member_id) for member_id in member_ids
+    ])
+
+    return {"members": members, "next_cursor": next_cursor}
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["channels:read", "groups:read", "im:read", "mpim:read"],
+    )
+)
+async def get_members_from_conversation_by_name(
```

Comment:
Tested with non-existing conversation name: "get the members in the slack conversation 'ggeneral'". Expected retryable tool error to correct it, but got the following instead.

Error: `Tool Slack.GetMembersFromConversationByName version 0.1.8 failed. Error in get_conversation_metadata_by_name: 'NoneType' object has no attribute 'lower'`

Reply:
fixed